src/metabase/models/card.clj

```diff
@@ -78,4 +80,44 @@
   revision/IRevisioned
   {:serialize-instance serialize-instance
    :revert-to-revision revision/default-revert-to-revision
-   :describe-diff      revision/default-describe-diff})
+   :diff-map           revision/default-diff-map
+   :diff-str           revision/default-diff-str})
+
+
+;;; ## ---------------------------------------- DEPENDENCIES ----------------------------------------
+
+
+(defn- parse-filter-subclause [subclause]
+  (match subclause
+    ["SEGMENT" (segment-id :guard integer?)] segment-id
+    subclause                                nil))
+
+(defn- parse-filter [clause]
+  (match clause
+    ["AND" & subclauses] (mapv parse-filter subclauses)
+    ["OR" & subclauses]  (mapv parse-filter subclauses)
+    subclause            (parse-filter-subclause subclause)))
+
+(defn- extract-segment-ids [filter-clause]
+  (when filter-clause
+    (->> (parse-filter filter-clause)
+         flatten
+         (filter identity)
+         set)))
+
+(defn- extract-metric-ids [aggregation-clause]
+  (match aggregation-clause
+    ["METRIC" (metric-id :guard integer?)] #{metric-id}
```

Comment:
Actually now that I think about it I'd say don't even bother. I played around with switching the QL to case-insensitivity for tokens like that. It was easy to implement and simplified a lot of code. So at some point in the near future I'll probably just merge that in